src/session.rs

```diff
@@ -75,36 +94,40 @@ impl<Connection: io::Read + io::Write + Sync + Send> Session<Connection> {
             Request::ShowPublicKey(ref req) => self.get_public_key(req)?,
             Request::PoisonPill(_req) => return Ok(false),
         };
-        //
+
         let mut buf = vec![];
+
         match response {
             Response::SignedHeartBeat(shb) => shb.encode(&mut buf)?,
             Response::SignedProposal(sp) => sp.encode(&mut buf)?,
             Response::SignedVote(sv) => sv.encode(&mut buf)?,
             Response::Ping(ping) => ping.encode(&mut buf)?,
             Response::PublicKey(pk) => pk.encode(&mut buf)?,
         }
+
         self.connection.write_all(&buf)?;
         Ok(true)
     }
 
     /// Perform a digital signature operation
     fn sign(&mut self, mut request: impl TendermintResponse) -> Result<Response, KmsError> {
         let mut to_sign = vec![];
-        // TODO(ismail): this should either be a config param, or, included in the request!
```

Comment:
FYI @Liamsi, looks like I managed to fix one of your TODOs by adding `chain::Id` configuration to the KMS's TOML file